applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -4,3 +4,409 @@
+	.started = stream_started_cb,
+	.metadata_updated = stream_metadata_updated_cb,
+	.disabled = stream_disabled_cb,
+	.stopped = stream_stopped_cb,
+	.released = stream_released_cb,
+};
+
+static void add_remote_sink(struct bt_audio_ep *ep, uint8_t index)
+{
+	if (index > sizeof(sinks)) {
+		LOG_ERR("Sink index is out of range");
+	} else {
+		sinks[index] = ep;
+	}
+}
+
+static void add_remote_codec(struct bt_codec *codec, int index, uint8_t type)
+{
+	if (type != BT_AUDIO_SINK && type != BT_AUDIO_SOURCE) {
+		return;
+	}
+
+	if (index < CONFIG_BT_AUDIO_UNICAST_CLIENT_PAC_COUNT) {
+		remote_codecs[index] = codec;
+	}
+}
+
+static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struct bt_audio_ep *ep,
+			     struct bt_audio_discover_params *params)
+{
+	int ret;
+
+	if (params->err) {
+		LOG_ERR("Discovery failed: %d", params->err);
+		return;
+	}
+
+	if (codec != NULL) {
+		add_remote_codec(codec, params->num_caps, params->type);
+		return;
+	}
+
+	if (ep != NULL) {
+		if (params->type == BT_AUDIO_SINK) {
+			add_remote_sink(ep, params->num_eps);
+		} else {
+			LOG_ERR("Invalid param type: %u", params->type);
+		}
+
+		return;
+	}
+
+	LOG_INF("Discover complete: err %d", params->err);
+
+	(void)memset(params, 0, sizeof(*params));
+	ret = bt_audio_stream_config(default_conn, &audio_stream, sinks[0],
+				     &lc3_preset_unicast_nrf5340.codec);
+	if (ret) {
+		LOG_ERR("Could not configure stream");
+	}
+}
+
+static int device_found(uint8_t type, const uint8_t *data, uint8_t data_len,
+			const bt_addr_le_t *addr)
+{
+	int ret;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+	if ((data_len == DEVICE_NAME_PEER_LEN) &&
+	    (strncmp(DEVICE_NAME_PEER, data, DEVICE_NAME_PEER_LEN) == 0)) {
+		LOG_INF("Device found");
+		ret = bt_le_scan_stop();
+		if (ret) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+		ret = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, BT_LE_CONN_PARAM_MULTI,
+					&default_conn);
+		if (ret) {
+			LOG_WRN("Create ACL connection failed: %d", ret);
+			ble_acl_start_scan();
+		}
+	}
+
+	return -ENOENT;
+}
+
+/** @brief  Parse BLE advertisement package.
+ */
+static void ad_parse(struct net_buf_simple *p_ad, const bt_addr_le_t *addr)
+{
+	while (p_ad->len > 1) {
+		uint8_t len = net_buf_simple_pull_u8(p_ad);
+		uint8_t type;
+
+		/* Check for early termination */
+		if (len == 0) {
+			return;
+		}
+
+		if (len > p_ad->len) {
+			LOG_INF("AD malformed");
+			return;
+		}
+
+		type = net_buf_simple_pull_u8(p_ad);
+
+		if (device_found(type, p_ad->data, len - 1, addr) == 0) {
+			return;
+		}
+
+		(void)net_buf_simple_pull(p_ad, len - 1);
+	}
+}
+
+static void on_device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			    struct net_buf_simple *p_ad)
+{
+	/* Note: May lead to connection creation */
+	ad_parse(p_ad, addr);
+}
+
+static void ble_acl_start_scan(void)
+{
+	int ret;
+
+	//TODO: Figure out why SCAN_PASSIVE leads to bad performance
```

Comment:
What do you mean by bad performance? The only thing that should change is the time it takes before a headset is found, once the connection is set up, everything should be the same

Reply:
I would assume the name from headset is sent through scan_rsp, because I found once if I set to passive scan, it is hard to find the device and connected.